server/resources/resource_management_test.go

```diff
@@ -2202,6 +2238,72 @@ var _ = Describe("Resource manager", func() {
+		}
+
+		Context("Custom action isolation tests", func() {
+			var NoProblem resources.ResourceProblem = -1
+
+			DescribeTable("Performing action", func(auth schema.Authorization, resourceID string, action *schema.Action,
+				expectedResourceProblem resources.ResourceProblem, shouldFail bool) {
+				addAuthAndEmptyResponseToContext(auth)
+				err := resources.ActionResource(
+					context, testDB, currentSchema, *action, resourceID,
+					map[string]interface{}{"test_string": "test string"})
+
+				if shouldFail {
+					Expect(err).To(HaveOccurred())
+					returnedErr, ok := err.(resources.ResourceError)
+					Expect(ok).To(BeTrue())
+					Expect(returnedErr.Problem).To(Equal(expectedResourceProblem))
+				} else {
+					Expect(err).NotTo(HaveOccurred())
+				}
+
+			},
+				Entry("Should not fail when performed action matched to policy with is_owner on own resource as member",
+					createAuthToken(tenantMemberAuthScope), memberResourceID, &singularAction, NoProblem, false),
+				Entry("Should not fail when performed action matched to policy with is_owner_domain on domain domain as domain admin",
+					createAuthToken(domainAdminAuthScope), adminResourceID, &singularAction, NoProblem, false),
+				Entry("Should not fail when performed action on different domain resource as cloud admin",
+					createAuthToken(superAdminAuthScope), otherDomainResourceID, &singularAction, NoProblem, false),
+				Entry("Should fail when performed action matched to policy with is_owner on not own resource as member",
+					createAuthToken(tenantMemberAuthScope), adminResourceID, &singularAction, resources.NotFound, true),
+				Entry("Should fail when performed action matched to policy with is_owner_domain on not own resource nor domain as domain admin",
+					createAuthToken(domainAdminAuthScope), otherDomainResourceID, &singularAction, resources.NotFound, true),
+
+				Entry("Should not fail when performed action matched to policy without is_owner on own resource as member",
+					createAuthToken(tenantMemberAuthScope), memberResourceID, &singularPublicAction, NoProblem, false),
+				Entry("Should not fail when performed action matched to policy without is_owner on domain domain as domain admin",
+					createAuthToken(domainAdminAuthScope), adminResourceID, &singularPublicAction, NoProblem, false),
+				Entry("Should not fail when performed action on different domain resource as cloud admin",
+					createAuthToken(superAdminAuthScope), otherDomainResourceID, &singularPublicAction, NoProblem, false),
+				Entry("Should not fail when performed action matched to policy without is_owner on not own resource as member",
+					createAuthToken(tenantMemberAuthScope), adminResourceID, &singularPublicAction, NoProblem, false),
+				Entry("Should not fail when performed action matched to policy without is_owner_domain on not own resource nor domain as domain admin",
+					createAuthToken(domainAdminAuthScope), otherDomainResourceID, &singularPublicAction, NoProblem, false),
```

Comment:
We only have one policy for `singular_public` and it's for `Member`s, why does this entry pass? Same goes for the other entry using domain admin.

Reply:
In the test_schema.yaml there is a policy:
```
- action: '*'
  effect: allow
  id: admin_statement
  principal: admin
  resource:
    path: .*
```

Reply:
Discussed offline, waiting for fix.

Reply:
I will not remove this policy to avoid breaking old tests, I have added extra policy for avoid confusing about this test case.